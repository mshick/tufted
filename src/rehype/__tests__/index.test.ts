import { html } from 'js-beautify'
import path from 'node:path'
import rehypeStringify from 'rehype-stringify'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import { readSync } from 'to-vfile'
import { unified } from 'unified'
import { expect, test } from 'vitest'
import preset from '../index.js'

test('the preset works', async () => {
  const processed = await unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeStringify)
    .use(preset({ assets: './', base: '/' }))
    .process(readSync(path.resolve(__dirname, '../../__tests__/example.md')))

  expect(html(String(processed))).toMatchSnapshot()
})
