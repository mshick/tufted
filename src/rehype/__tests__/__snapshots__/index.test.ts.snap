// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`the preset works 1`] = `
"<h1 id="title-skip-for-initial-heading">Title Skip for Initial Heading<a class="heading-link hidden" href="#title-skip-for-initial-heading"><span class="icon icon-link"></span></a></h1>
<p>Introductory text without heading will add a hidden h2 added. This allows for a
    consistent TOC. Consectetur adipiscing elit, sed do eiusmod tempor incididunt
    ut <code>inline code</code> et dolore magna aliqua. Vestibulum rhoncus est pellentesque
    elit. Cras sed felis eget velit. Accumsan in nisl nisi scelerisque eu.</p>
<p>This YouTube directive will be upgraded to a figure-wrapped iframe.</p>
<p>::youtube[TOTINO BOY!!]{#2lPX5b9m7ro}</p>
<p>Vivamus at augue eget arcu dictum varius. In cursus turpis massa tincidunt dui
ut. Tincidunt ornare massa eget egestas purus viverra. Ultricies tristique
nulla aliquet enim tortor at auctor urna nunc. Nec ullamcorper sit amet risus
nullam. Morbi tempus iaculis urna id. Sed turpis tincidunt id aliquet risus
feugiat in ante metus.[^1] Integer vitae justo eget magna fermentum iaculis.
Libero enim sed faucibus turpis. Ut placerat orci nulla pellentesque dignissim
enim sit amet. In eu mi bibendum neque. Blandit massa enim nec dui nunc.
Posuere sollicitudin aliquam ultrices sagittis orci a scelerisque. Imperdiet
proin fermentum leo vel orci porta non. Suspendisse in est ante in nibh. Eget
sit amet tellus cras adipiscing enim.</p>
<p>[^1]: <a href="https://www.google.com">Referenced Footnote To Sidenote</a></p>
<h2 id="heading-starts-section">Heading Starts Section<a class="heading-link hidden" href="#heading-starts-section"><span class="icon icon-link"></span></a></h2>
<h3 id="subheading-no-section">Subheading No Section<a class="heading-link hidden" href="#subheading-no-section"><span class="icon icon-link"></span></a></h3>
<p>You can create explicit figures. This allows for optional classes, like
<code>.fullwidth</code>. It also makes it easy to include a <code>figcaption</code>.</p>
<p>:::figure{.fullwidth}</p>
<p><img src="https://picsum.photos/200/300" alt="Directive wrapped figure"></p>
<p>Sagittis nisl rhoncus mattis rhoncus urna neque viverra justo nec. Egestas
purus viverra accumsan in nisl nisi scelerisque. Eu mi bibendum neque egestas
congue.</p>
<p>:::</p>
<p>Figure content and caption text can also be flipped. And, a figure can describe
a YouTube video.</p>
<p>:::figure{.fullwidth}</p>
<p>Sagittis nisl rhoncus mattis rhoncus urna neque viverra justo nec. Egestas
purus viverra accumsan in nisl nisi scelerisque. Eu mi bibendum neque egestas
congue.</p>
<p>::youtube[Pizza Pizza]{#2lPX5b9m7ro}</p>
<p>:::</p>
<h2 id="sidenotes">Sidenotes<a class="heading-link hidden" href="#sidenotes"><span class="icon icon-link"></span></a></h2>
<p>Diam phasellus vestibulum lorem sed risus ultricies tristique. Arcu odio ut sem
nulla pharetra diam sit amet. At augue eget arcu dictum. Viverra ipsum nunc
aliquet bibendum enim.[^margin] Magna eget est lorem ipsum dolor. Malesuada nunc vel
risus commodo viverra maecenas accumsan lacus. Tellus orci ac auctor augue
mauris. Lorem ipsum dolor sit amet consectetur. Viverra aliquet eget sit amet
tellus cras adipiscing enim. Amet justo donec enim diam vulputate ut pharetra
sit. Accumsan sit amet nulla facilisi morbi tempus iaculis urna. Eu nisl nunc
mi ipsum faucibus vitae aliquet nec. Quam nulla porttitor massa id neque
aliquam vestibulum morbi. Leo vel orci porta non pulvinar neque. In mollis
nunc sed id semper risus. Massa tempor nec feugiat nisl pretium fusce id velit
ut. Eu tincidunt tortor aliquam nulla facilisi cras fermentum odio. Turpis
tincidunt id aliquet risus. Faucibus scelerisque eleifend donec pretium
vulputate sapien nec. Turpis in eu mi bibendum neque egestas congue.</p>
<p>[^margin]:
<img src="./local-image.png" alt="Image in a marginnote"> At urna condimentum mattis
pellentesque id nibh. Nulla facilisi nullam vehicula ipsum a. Diam quam
nulla porttitor massa id neque aliquam vestibulum.</p>
<p><img src="./local-image.png" alt="Standalone image gets wrapped in figure"></p>
<p>Amet mauris commodo quis imperdiet massa tincidunt. Adipiscing elit ut aliquam
purus sit amet luctus venenatis. Odio tempor orci dapibus ultrices in iaculis
nunc sed. Viverra maecenas accumsan lacus vel facilisis volutpat. Nunc id
cursus metus aliquam eleifend mi in nulla posuere. Vitae tempus quam
pellentesque nec. Enim neque volutpat ac tincidunt vitae semper quis.
Sit amet luctus venenatis lectus magna fringilla.</p>
<blockquote>
<p>This blockquote has a footer that contains a footnote citation. Tellus rutrum
tellus pellentesque eu tincidunt tortor. Tincidunt id aliquet risus feugiat
in. Elementum integer enim neque volutpat ac tincidunt vitae.</p>
<p>::footer[<a href="https://www.reddit.com/r/AskHistorians/comments/1l2jge/how_significant_was_cicero_in_the_events_that_led/">Marcus Tullius Cicero, reddit, 'How To Fill Text' subreddit</a>]</p>
</blockquote>
<p>:newthought[Rhoncus urna neque] viverra justo. Sed id semper risus in hendrerit
gravida. Diam vel quam elementum pulvinar etiam. Lacus sed turpis tincidunt id
aliquet risus feugiat in ante. Dui vivamus arcu felis bibendum ut tristique et
egestas. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula
ullamcorper. Feugiat vivamus at augue eget. Amet purus gravida quis blandit
turpis cursus in. Fermentum leo vel orci porta non pulvinar neque laoreet.</p>
<h2 id="code">Code<a class="heading-link hidden" href="#code"><span class="icon icon-link"></span></a></h2>
<p>Code blocks support titles. Titles are typically filenames, and they cannot
contain spaces.</p>
<pre class="shiki shiki-themes vitesse-light vitesse-dark" style="background-color:#ffffff;--shiki-dark-bg:#121212;color:#393a34;--shiki-dark:#dbd7caee" tabindex="0"><code><span class="line"><span style="color:#AB5959;--shiki-dark:#CB7676">function</span><span style="color:#59873A;--shiki-dark:#80A665"> randomCode</span><span style="color:#999999;--shiki-dark:#666666">(</span><span style="color:#B07D48;--shiki-dark:#BD976A">bar</span><span style="color:#999999;--shiki-dark:#666666">)</span><span style="color:#999999;--shiki-dark:#666666"> {</span></span>
<span class="line"><span style="color:#AB5959;--shiki-dark:#CB7676">  var</span><span style="color:#B07D48;--shiki-dark:#BD976A"> a</span><span style="color:#999999;--shiki-dark:#666666"> =</span><span style="color:#2F798A;--shiki-dark:#4C9A91"> 42</span><span style="color:#999999;--shiki-dark:#666666">;</span></span>
<span class="line"><span style="color:#AB5959;--shiki-dark:#CB7676">  var</span><span style="color:#B07D48;--shiki-dark:#BD976A"> b</span><span style="color:#999999;--shiki-dark:#666666"> =</span><span style="color:#B5695999;--shiki-dark:#C98A7D99"> '</span><span style="color:#B56959;--shiki-dark:#C98A7D">Prism</span><span style="color:#B5695999;--shiki-dark:#C98A7D99">'</span><span style="color:#999999;--shiki-dark:#666666">;</span></span>
<span class="line"><span style="color:#1E754F;--shiki-dark:#4D9375">  return</span><span style="color:#B07D48;--shiki-dark:#BD976A"> a</span><span style="color:#AB5959;--shiki-dark:#CB7676"> +</span><span style="color:#59873A;--shiki-dark:#80A665"> bar</span><span style="color:#999999;--shiki-dark:#666666">(</span><span style="color:#B07D48;--shiki-dark:#BD976A">b</span><span style="color:#999999;--shiki-dark:#666666">);</span></span>
<span class="line"><span style="color:#999999;--shiki-dark:#666666">}</span></span>
<span class="line"></span></code></pre>
<h2 id="epigraphs">Epigraphs<a class="heading-link hidden" href="#epigraphs"><span class="icon icon-link"></span></a></h2>
<blockquote>
<p>We get wrapped in an epigraph tag. Vel pharetra vel turpis nunc eget lorem
dolor sed. Ac odio tempor orci dapibus. Ac odio tempor orci dapibus ultrices
in iaculis.</p>
<p>::footer[Marc Cicero, "De finibus bonorum et malorum"^[^footer]]</p>
</blockquote>
<blockquote>
<p>Amet consectetur adipiscing elit duis tristique sollicitudin nibh. Luctus
accumsan tortor posuere ac ut consequat semper. Est ante in nibh mauris
cursus mattis molestie a.</p>
<p>::footer[Marc Cicero, :cite["Using Default Remark Directive Citation"]]</p>
</blockquote>
<p>Amet luctus venenatis lectus magna fringilla urna porttitor rhoncus. Amet est
placerat in egestas erat imperdiet. Volutpat consequat mauris nunc congue nisi
vitae suscipit tellus. Viverra maecenas accumsan lacus vel facilisis volutpat
est. Augue ut lectus arcu bibendum at varius. Ut pharetra sit amet aliquam id
diam maecenas ultricies.</p>
<p>[^footer]: An inline note in the blockquote footer</p>
<h2 id="tables">Tables<a class="heading-link hidden" href="#tables"><span class="icon icon-link"></span></a></h2>
<p>Table processing in <code>remark-gfm</code>. Enim tortor at auctor urna nunc id cursus.
Sit amet commodo nulla facilisi. Laoreet non curabitur gravida arcu ac tortor
dignissim. Elementum tempus egestas sed sed risus pretium quam vulputate
dignissim. Tortor aliquam nulla facilisi cras fermentum.</p>
<p>Colons can be used to align columns.</p>
<p>| Tables        |      Are      |   Cool |
| ------------- | :-----------: | -----: |
| col 3 is      | right-aligned | $1600 |
| col 2 is      |   centered    |   $12 |
| zebra stripes |   are neat    |    $1 |</p>
<p>There must be at least 3 dashes separating each header cell. The outer pipes (|)
are optional, and you don't need to make the raw Markdown line up prettily. You
can also use inline Markdown.</p>
<p>| Markdown | Less      | Pretty     |
| -------- | --------- | ---------- |
| <em>Still</em>  | <code>renders</code> | <strong>nicely</strong> |
| 1        | 2         | 3          |</p>
<h2 id="lists">Lists<a class="heading-link hidden" href="#lists"><span class="icon icon-link"></span></a></h2>
<p>Quam id leo in vitae turpis massa sed. Sagittis id consectetur purus ut
faucibus pulvinar elementum. Magna etiam tempor orci eu lobortis elementum.</p>
<p>An unordered list:</p>
<ul>
<li>Quam</li>
<li>lacus</li>
<li>suspendisse</li>
<li>faucibus</li>
</ul>
<p>An ordered list:</p>
<ol>
<li>Eget</li>
<li>magna</li>
<li>fermentum</li>
<li>iaculis</li>
</ol>"
`;
